### DISCUSSION TITLE
Supporting Wildcards in `staticwebapp.config.json`

### DISCUSSION THREAD

**Discussion by brionmario** [Azure/static-web-apps#1481] [General]
I'm looking for suggestions to implement custom routing logic in the `staticwebapp.config.json` file. I need to map various URL patterns to specific HTML files, as described below:

```pug
/app* -> /app/index.html
/t/*/-> /index.html
/t/*/app*-> app/index.html
/o/*/ -> /index.html
/o/*/app* -> /app/index.html

fallback: index.html
```

This is what i've tried so far:

```json
{
  "routes": [
    {
      "route": "/app*",
      "rewrite": "/app/index.html"
    },
    {
      "route": "/t/*/",
      "rewrite": "/index.html"
    },
    {
      "route": "/t/*/app*",
      "rewrite": "/app/index.html"
    },
    {
      "route": "/o/*/",
      "rewrite": "/index.html"
    },
    {
      "route": "/o/*/app*",
      "rewrite": "/app/index.html"
    }
  ],
  "navigationFallback": {
    "rewrite": "/index.html"
  }
}
```

This seems to break due to having two `*`.

Any suggestions would be highly appreciated

**Comment by johnnyreilly** [comment]
You might find that the approach I've documented here solves your problem 

https://johnnyreilly.com/azure-static-web-apps-dynamic-redirects-azure-functions

**Reply by brionmario** [reply]
Thanks @johnnyreilly. I'll check this out.

**Reply by OzBob** [reply]
@johnnyreilly in 2022 I refered to your solution in the open GitHub issue here https://github.com/Azure/static-web-apps/discussions/1481 that @ShanonJackson @anthonychu also tweeted.

but ... it's a redirection solution, not a rewrite.

@mkarmark said 'We currently don't support wildcard paths for rewrite, but can consider supporting that in the future!"

is there a way to 'vote up' the feature request?

**Reply by johnnyreilly** [reply]
I don't think there is I'm afraid
